src/reason_parser.mly

```diff
@@ -788,8 +788,28 @@ let rec string_of_longident = function
 
 let built_in_explicit_arity_constructors = ["Some"; "Assert_failure"; "Match_failure"]
 
+let jsx_component module_name attrs children ~loc =
+  let firstPart = (List.hd (Longident.flatten module_name)) in
+  let lident = if firstPart = String.capitalize firstPart then
+    Ldot(module_name, "createElement")
+  else
+    Lident firstPart
+  in
+  let ident = ghloc ~loc lident in
+  let body = mkexp(Pexp_apply(mkexp(Pexp_ident ident), attrs @ children)) in
```

Comment:
You also want to pass loc here when you call mkexp (Pexp_ident ident). Otherwise, you will generate an AST node without location information and our IDE support will get confused. 

Currently, with your diff, formatting a code of: 
```
let a = <Foo a />;
```
gives us
```
[
  structure_item (try.re[1,0+0]..[1,0+18])
    Pstr_value Nonrec
    [
      <def>
        pattern (try.re[1,0+4]..[1,0+5])
          Ppat_var "a" (try.re[1,0+4]..[1,0+5])
        expression (try.re[1,0+8]..[1,0+17])
          attribute "JSX"
            []
          Pexp_apply
          expression ([0,0+-1]..[0,0+-1])
            Pexp_ident "Foo.createElement" (try.re[1,0+8]..[1,0+17]) ghost
          [
            <label> "a"
              expression ([0,0+-1]..[0,0+-1])
                Pexp_ident "a" (try.re[1,0+13]..[1,0+14])
            <label> ""
              expression (try.re[1,0+8]..[1,0+17]) ghost
                Pexp_construct "[]" (try.re[1,0+8]..[1,0+17]) ghost
                None
          ]
    ]
]
```

(You can check the generated AST by:
```
refmt -parse re -print ast try.re
```
)